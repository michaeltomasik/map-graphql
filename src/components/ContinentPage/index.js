import React, { Component, Fragment } from 'react'
import { Query } from 'react-apollo'
import  { gql } from 'apollo-boost'
import { ListGroup } from 'react-bootstrap'

import GraphQLResultAwait from '../GraphQLResultAwait'
import './ContinentPage.css'

export default class ContinentPage extends Component {
  goToCountriesList(continentCode) {
    this.props.history.push(`/articles/${continentCode}`);
  }

  render() {
    return (
      <Query query={CONTINENT_QUERY} variables={{ continentCode: this.props.match.params.continentCode }}>
        {({ data, loading, error, refetch }) => {
          if (loading || error) {
            return <GraphQLResultAwait loading={loading} error={error} />
          }

          return (
            <Fragment>
              <h1>Country: {data.getContinent.name}</h1>
              <ListGroup>{
                data.getContinent.countries.map(countryObj =>
                  <ListGroup.Item
                    action
                    as="li"
                    variant="info"
                    key={countryObj.name}
                    className="country"
                    onClick={() => { this.goToCountriesList(countryObj.name) }}>{countryObj.name}</ListGroup.Item>)
              }</ListGroup>
            </Fragment>
          )
        }}
      </Query>
    )
  }
}

export const CONTINENT_QUERY = gql`
  query Continent($continentCode: String) {
    getContinent(continentCode: $continentCode){
        name
        countries {
          name,
        }
    }
  }
`