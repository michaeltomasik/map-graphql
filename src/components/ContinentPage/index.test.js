import React from 'react'
import { MockedProvider } from 'react-apollo/test-utils';
import { mount } from 'enzyme';
// The component AND the query need to be exported
import ContinentPage, { CONTINENT_QUERY } from './index';
import wait from 'waait';

const mocks = [
  {
    request: {
      query: CONTINENT_QUERY,
      variables: {
        continentCode: 'EU',
      },
    },
    result: {
      data: {
        getContinent: {
          name: 'EU',
          countries: [
            {
              "name": "Nepal",
              "code": "NP",
              "native": "à¤¨à¤ªà¤²",
              "emoji": "ðŸ‡³ðŸ‡µ"
            },
            {
              "name": "Oman",
              "code": "OM",
              "native": "Ø¹Ù…Ø§Ù†",
              "emoji": "ðŸ‡´ðŸ‡²"
            },
          ]
        },
      },
    },
  },
];

const props = {
  match: {
    params: {
      continentCode: 'EU',
    }
  }
}

describe('ContinentPage form test suite', () => {
  it('renders with data', async () => {
    const wrapper = mount((
      <MockedProvider mocks={mocks} addTypename={false}>
        <ContinentPage {...props} />
      </MockedProvider>
    ));
    // https://github.com/apollographql/react-apollo/issues/2719
    await wait(0);
    wrapper.update()

    expect(wrapper.debug()).toMatchSnapshot();
  });
});