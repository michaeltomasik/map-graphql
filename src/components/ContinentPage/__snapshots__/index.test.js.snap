// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContinentPage form test suite renders with data 1`] = `
"<MockedProvider mocks={{...}} addTypename={false}>
  <ApolloProvider client={{...}}>
    <ContinentPage match={{...}}>
      <Query query={{...}} variables={{...}}>
        <h1>
          Country: 
          EU
        </h1>
        <ForwardRef>
          <Uncontrolled(Bootstrap(ListGroup)) innerRef={{...}}>
            <Bootstrap(ListGroup) activeKey={[undefined]} onSelect={[Function: handleChange]}>
              <ListGroup activeKey={[undefined]} onSelect={[Function: handleChange]} bsPrefix=\\"list-group\\" variant={{...}}>
                <ContextTransform(AbstractNav) activeKey={[undefined]} onSelect={[Function: handleChange]} as=\\"div\\" className=\\"list-group\\">
                  <AbstractNav activeKey={[undefined]} onSelect={[Function: handleChange]} as=\\"div\\" className=\\"list-group\\" parentOnSelect={[undefined]}>
                    <div className=\\"list-group\\" onKeyDown={[Function]}>
                      <Bootstrap(ListGroupItem) action={true} as=\\"li\\" variant=\\"info\\" className=\\"country\\" onClick={[Function: onClick]}>
                        <ListGroupItem action={true} as=\\"li\\" variant=\\"info\\" className=\\"country\\" onClick={[Function: onClick]} bsPrefix=\\"list-group-item\\" active={false} disabled={false}>
                          <ForwardRef onClick={[Function]} eventKey={{...}} as=\\"li\\" className=\\"country list-group-item list-group-item-info list-group-item-action\\" disabled={false}>
                            <li disabled={false} data-rb-event-key={{...}} id={[undefined]} aria-controls={[undefined]} onClick={[Function]} className=\\"country list-group-item list-group-item-info list-group-item-action\\">
                              Nepal
                            </li>
                          </ForwardRef>
                        </ListGroupItem>
                      </Bootstrap(ListGroupItem)>
                      <Bootstrap(ListGroupItem) action={true} as=\\"li\\" variant=\\"info\\" className=\\"country\\" onClick={[Function: onClick]}>
                        <ListGroupItem action={true} as=\\"li\\" variant=\\"info\\" className=\\"country\\" onClick={[Function: onClick]} bsPrefix=\\"list-group-item\\" active={false} disabled={false}>
                          <ForwardRef onClick={[Function]} eventKey={{...}} as=\\"li\\" className=\\"country list-group-item list-group-item-info list-group-item-action\\" disabled={false}>
                            <li disabled={false} data-rb-event-key={{...}} id={[undefined]} aria-controls={[undefined]} onClick={[Function]} className=\\"country list-group-item list-group-item-info list-group-item-action\\">
                              Oman
                            </li>
                          </ForwardRef>
                        </ListGroupItem>
                      </Bootstrap(ListGroupItem)>
                    </div>
                  </AbstractNav>
                </ContextTransform(AbstractNav)>
              </ListGroup>
            </Bootstrap(ListGroup)>
          </Uncontrolled(Bootstrap(ListGroup))>
        </ForwardRef>
      </Query>
    </ContinentPage>
  </ApolloProvider>
</MockedProvider>"
`;
